frontend/src/layouts/AppLayout.jsx

```diff
@@ -1,9 +1,25 @@
-import { Outlet } from "react-router-dom";
+import { useContext, useEffect, useState } from "react";
+import { Navigate, Outlet } from "react-router-dom";
+import { AuthContext } from "../contexts/AuthContext";
 
 export default function AppLayout() {
+  const token = "";
+  const { checkAuth } = useContext(AuthContext);
+  const [isAuthUrl, setIsAuthUrl] = useState(null);
+
+  useEffect(() => {
+    const navigateURL = checkAuth(token);
+    if (navigateURL) {
+      setIsAuthUrl(navigateURL);
+    }
+  }, [token, checkAuth]);
+  if (isAuthUrl) {
+    return <Navigate to={isAuthUrl} />;
+  }
```

Comment:
This code should be updated too. functionality of checking authorization, and redirecting to login when user is unauthenticated, should be moved into a custom hook.